### ISSUE TITLE
SSM Package fileinput does not accept 'mixed' keyword MF6

### ISSUE THREAD

**Issue by laat0003** [modflowpy/flopy#1909]
Howdy,
Unsure if this is a bug or intended so apologies upfront for being annoying.

**Describe the bug**
SSM package throws an error if you try and use the keyword 'mixed' in the fileinput 

**To Reproduce**
try something like this:

ssm = flopy.mf6.ModflowGwtssm(gwt, fileinput=[('rch',"trans_1spc.spca"), ("evt", "trans_2spc.spca", "MIXED")]) 

and you will get this:

(1) An error occurred when storing data "fileinput" in a recarray. fileinput data is a one or two dimensional list containing the variables "<pname> <spc6_filename>" (some variables may be optional, see MF6 documentation), but data "[('rch', 'trans_1spc.spca'), ('evt', 'trans_2spc.spca', 'MIXED')]" was supplied.

**Expected behavior**
This works and is what I thought was supposed to be happening with the original constructor.
ssm = flopy.mf6.ModflowGwtssm(gwt, fileinput=[('rch',"trans_1spc.spca")])
ssm.fileinput.append_list_as_record(["evt", "trans_2spc.spca",  "MIXED"])

**Desktop (please complete the following information):**
 - OS: Windows 11
 -Flopy version 3.4.1

Great work as always.
Cheers,
Tariq

**Comment by langevin-usgs** [comment]
I've verified that flopy is having trouble with the MIXED keyword.  @spaulins-usgs, can you take a look?  You can easily reproduce this issue with a MODFLOW 6 autotest [here](https://github.com/MODFLOW-USGS/modflow6/blob/5d4f56a5c785c9dd5681032f2a6c8b766533d9af/autotest/test_gwt_ssm04.py#L285).  Try adding the keyword "MIXED" to any of these lines in the FILEINPUT block (as shown below), and flopy will give the error reported by @laat0003.

```
    # ssm package
    sourcerecarray = [()]
    fileinput = [
        ("RCH-1", f"{gwtname}.rch1.spc"),
        ("RCH-2", f"{gwtname}.rch2.spc"),
        ("RCH-3", f"{gwtname}.rch3.spc", "MIXED"),
        ("RCH-4", f"{gwtname}.rch4.spc"),
    ]
    ssm = flopy.mf6.ModflowGwtssm(
        gwt, print_flows=True, sources=sourcerecarray, fileinput=fileinput
    )
```

**Comment by spaulins-usgs** [comment]
@laat0003 please try the latest flopy code from the develop branch.  It has a fix that should solve this problem.

**Comment by wpbonelli** [comment]
removing from 3.4.3 milestone as this was closed by #1920 which was included in 3.4.2
